boot/bootutil/src/bootutil_misc.c

```diff
@@ -248,6 +248,27 @@ boot_enc_key_off(const struct flash_area *fap, uint8_t slot)
 }
 #endif
 
+bool bootutil_buffer_is_erased(const struct flash_area *area,
```

Comment:
For me this name is not descriptive, maybe you can consider:
`bootutil_read_check_buff_is_erased()`

Reply:
Why?  It doesn't read anything.  It just checks the buffer against the erased value for the flash area specified.

Reply:
I do agree some documentation explaining what this is *intended* to do is important.

Reply:
right

Reply:
It could be `bootutil_check_buff_is_erased`, although if I had to write `buff` I would just use `buf` instead. :-P

I like the current naming, it feels quite clear already, and I added a comment header explaining what it does and returns.

Reply:
It's ok now